src/tests/tests.h

```diff
@@ -757,14 +757,18 @@ class TestFnRegistration {
       }
 };
 
+#define BOTAN_REGISTER_TEST_FN_IMPL(category, name, smoke_test, needs_serialization, fn0, ...) \
+   static const TestFnRegistration register_##fn0(                                             \
+      category, name, smoke_test, needs_serialization, {__FILE__, __LINE__}, fn0 __VA_OPT__(, ) __VA_ARGS__)
```

Comment:
Disambiguates the `static const TestFnRegistration` variable names by incorporating the first registered function name (`fn0`) into the variable identifier.

As a result, one can now do things like:

```C++

Test::Result test_sha1() {/* ...*/}

Test::Result test_sha256() {/* ...*/}
Test::Result test_sha384() {/* ...*/}
Test::Result test_sha512() {/* ...*/}

BOTAN_REGISTER_TEST_FN("hash", "sha1", test_sha1);
BOTAN_REGISTER_TEST_FN("hash", "sha2", test_sha256, test_sha384, test_sha512);
```

Reply:
Its a shame Clang’s `__COUNTER__` isnt standard

Reply:
Today I learnt! Googling that, and someone suggests to use `__LINE__` instead. And in fact that would actually work just fine for this use case. Good to keep in mind, perhaps.